flask_restful/inputs.py

```diff
@@ -278,3 +279,18 @@ def datetime_from_iso8601(datetime_str):
     :return: A datetime
     """
     return aniso8601.parse_datetime(datetime_str)
+
+
+def datetime_from_rfc3339(datetime_str):
+    """Turns an RFC3339 formatted date into a UTC datetime object.
+
+    Example::
+
+        inputs.datetime_from_rfc3339("2013-03-25T12:42:31+00:32")
+
+    :param datetime_str: The RFC3339-complying string to transform
+    :type datetime_str: str
+    :return: A datetime
+    """
+    ts = strict_rfc3339.rfc3339_to_timestamp(datetime_str)
+    return datetime.fromtimestamp(ts, tz=pytz.utc)
```

Comment:
It will be the supplied time as UTC. One of the goals of strict RFC3339 is to "Avoid timezones as much as possible." Which is why these functions deal with UTC so much. It's much harder to shoot yourself in the foot if you run a strictly formatted date as input through this module and deal with UTC times everywhere.

```python
import pytz
import datetime
import strict_rfc3339

datetime_str1 = "2015-08-18T17:59:07.435698Z"
datetime_str2 = "2015-08-18T17:59:07.435698+02:00"

ts1 = strict_rfc3339.rfc3339_to_timestamp(datetime_str1)
ts2 = strict_rfc3339.rfc3339_to_timestamp(datetime_str2)

print(datetime.datetime.fromtimestamp(ts1, tz=pytz.utc))
print(datetime.datetime.fromtimestamp(ts2, tz=pytz.utc))
```

```
2015-08-18 17:59:07.435698+00:00
2015-08-18 15:59:07.435698+00:00
```